arch/rv32i/src/pmp.rs

```diff
@@ -411,7 +418,7 @@ impl kernel::mpu::MPU for PMP {
             return None;
         }
 
-        let region = PMPRegion::new(region_start as *const u8, region_size, permissions);
+        let region = PMPRegion::new(region_start as *const u8, initial_app_memory_size, permissions);
```

Comment:
This doesn't seem right, we should be using `region_size`.

Maybe instead the fix would be:
```diff
diff --git a/arch/rv32i/src/pmp.rs b/arch/rv32i/src/pmp.rs
index bfa682d03..7ab539b99 100644
--- a/arch/rv32i/src/pmp.rs
+++ b/arch/rv32i/src/pmp.rs
@@ -393,7 +393,7 @@ impl kernel::mpu::MPU for PMP {
         // Make sure there is enough memory for app memory and kernel memory.
         let mut region_size = cmp::max(
             min_memory_size,
-            initial_app_memory_size + initial_kernel_memory_size,
+            initial_app_memory_size,
         ) as usize;
 
         // Region size always has to align to 4 bytes
```
Note that the same fix is required in the ARM MPU.

Reply:
The returned `region_size` has to be the entire app's memory, not just the accessible portion. The accessible region is defined by the initial app memory size.

Reply:
That doesn't seem to be what the ARM MPU does though.

Reply:
It uses subregions to restrict access beyond the initial app break, and has to return the entire region size due to power of 2 restrictions.

I tried the same test on hail and it fails as expected.

Reply:
Ok, looks fine then.